### DISCUSSION TITLE
How can I trigger a separate workflow in another repo, without it affecting the main build pipeline?

### DISCUSSION THREAD

**Discussion by Leramir1** [community/community#139806] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Ex.:
1. Build Application  ~ 3 mins
2. Run Unit Tests ~ 3 mins
3. Deploy Application ~3 mins
- **If deployed successfully** -> **trigger separate workflow that runs extensive tests** ~ 2 hours
4. Pipeline stops reports it took ~9 mins.
  
My goal is to have the "build workflow" and the "regression testing workflow" operate independently, whether in separate pipelines or within the same pipeline. The key aspect being the "regression testing workflow" should be triggered automatically after the deployment is complete, without impacting the building, unit testing, or deployment processes of the application.
![image](https://github.com/user-attachments/assets/bfb86e74-df3f-49dc-9499-9b435b578d09)

**Comment by AbderrahmenISSA** [comment]
Hello @Leramir1 , have you got an answer for your question ? 
We have the same issue with a terraform repository to trigger after building a docker image.

**Comment by Leramir1** [comment, marked as answer]
Hi @AbderrahmenISSA, yes, I have! It all starts with GitHub workflow triggers:

- [workflow_dispatch](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_dispatch)

- [workflow_run](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_run)

I've identified two key scenarios where automatically triggering a separate workflow (pipeline) differs:

1. Passing information from the build to a child workflow (**workflow_dispatch**).

2. Triggering a child workflow without needing to pass information (**workflow_run**).

Using **workflow_dispatch**

- This approach is useful when passing details from the build into the testing workflow (e.g., version number).

- A workflow can be triggered via workflow_dispatch using a curl request: [Create a workflow dispatch event](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event).

    - Keep in mind there's a limit of 10 inputs that can be passed.

- To simplify this, I created a GitHub composite action instead of using raw curl commands. Alternatively, you can leverage existing solutions from the GitHub Marketplace, such as [Trigger workflow dispatch and wait](https://github.com/marketplace/actions/workflow-dispatch-and-wait).

- <img src="https://github.com/user-attachments/assets/d25f8cad-0cd5-417d-b3f2-3f936a1b0d5a" width="500">


Using **workflow_run**

- If no information needs to be passed from the build:

    - Specify the **Name:** of the Build Workflow in the Testing Workflow to automatically trigger once the build workflow is completed.

    - Listen for the action type: `completed`.

- <img src="https://github.com/user-attachments/assets/d8296827-540d-4313-bd3b-b1259c19778c" width="500">

- Note: **workflow_run** only triggers for completed workflows in the same repository.
